crates/vm/src/system/memory/online.rs

```diff
@@ -690,6 +691,15 @@ impl TracingMemory {
         pointer: usize,
         prev_values: &[T; BLOCK_SIZE],
     ) -> u32 {
+        debug_assert_eq!(ALIGN, self.data.memory.config[address_space].min_block_size);
+        let cell_size = unsafe {
```

Comment:
better if you make this a debug assert and then the compiler still uses `size_of::<T>()`